src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -345,6 +348,11 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			handler.EnqueueRequestsFromMapFunc(r.reconcileClusterForPods),
 			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
 		).
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
+			handler.EnqueueRequestsFromMapFunc(r.reconcileClusterForExternalCASecret),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+		).
```

Comment:
Is this needed? As it's been explained to me, seastar will automatically reload changes made to certificates. Since this is a mounted secret, it is supposed to be picked up automatically and no reconciliation of Cluster resources should be necessary. Is there some need for reconciliation that I'm misunderstanding?

Reply:
The secret storing the CA cert provided by customer is not the secret mounted to RP pod. A new secret is created and mounted to RP pod.  This new secret store the customer CA cert along with other trusted CA certs, e.g. node, see other CA certs at https://github.com/paulzhang97/redpanda/blob/9cba2d30df4b4f15d75d7ff0fdb4370d24d0016c/src/go/k8s/pkg/resources/certmanager/type_helpers.go#L340-L344

The watch is to watch the updates to the secret set in the Cluster CR.